gc/base/MemorySubSpaceSemiSpace.cpp

```diff
@@ -834,6 +834,7 @@ MM_MemorySubSpaceSemiSpace::checkSubSpaceMemoryPostCollectResize(MM_EnvironmentB
 {
 	MM_GCExtensionsBase *extensions = MM_GCExtensionsBase::getExtensions(env->getOmrVM());
 	uintptr_t regionSize = extensions->getHeap()->getHeapRegionManager()->getRegionSize();
+	MM_Scavenger *scavenger = (MM_Scavenger *)_collector;
```

Comment:
We should try to minimize dependency of SubSpace on a specific collector (imagine we have an alternative implementation of a collector under this same SubSpace, and in runtime we should be able to choose one or the other).

So, we could pass those timestamps as parameters, instead of fetching directly from the collector class.

In similar light, I'd rename _lastScavengeEndTime to more generic _lastGCEndTime.

Reply:
>We should try to minimize dependency of SubSpace on a specific collector (imagine we have an alternative implementation of a collector under this same SubSpace, and in runtime we should be able to choose one or the other). So, we could pass those timestamps as parameters, instead of fetching directly from the collector class.

Enson looked into this and it required a considerable amount of rework to the api calls, at the end we decided it would be better to forgo those changes as part of this PR. We felt it was fine given that SemiSpace code already relies heavily on Scavenger